src/backend/distributed/operations/auto_shard_split.c

```diff
@@ -52,90 +51,110 @@
+    int rowCount = SPI_processed;
+    int64** shardInfo = (int64**)palloc(rowCount * sizeof(int64*));
+    for (int i = 0; i < rowCount; i++) {
+        shardInfo[i] = (int64*)palloc(5 * sizeof(int64));
+    }
+    
+    int64 jobId = CreateBackgroundJob("auto split", "Automatic Split Shards having Size greater than threshold");
 
     for (int rowIndex = 0; rowIndex < rowCount; rowIndex++)
 	{
-		char *shardIdString = PQgetvalue(result, rowIndex, 1);
-		uint64 shardId = strtou64(shardIdString, NULL, 10);
+        HeapTuple tuple = tupletable->vals[rowIndex];
 
-		char *sizeString = PQgetvalue(result, rowIndex, 5);
-		uint64 totalSize = strtou64(sizeString, NULL, 10);
+        Datum shardId = SPI_getbinval(tuple,tupletable->tupdesc,1,&isnull);
+        shardInfo[rowIndex][0]= DatumGetInt64(shardId);
 
-        char *nodeIdString = PQgetvalue(result, rowIndex, 6);
-        uint64 nodeId = strtou64(nodeIdString, NULL, 10);
+		Datum shardSize = SPI_getbinval(tuple,tupletable->tupdesc,4,&isnull);
+        shardInfo[rowIndex][1]= DatumGetInt64(shardSize);
 
-        char *minValueString = PQgetvalue(result, rowIndex, 3);
-		uint64 shardMinValue = strtou64(minValueString, NULL, 10);
+		Datum nodeId = SPI_getbinval(tuple,tupletable->tupdesc,5,&isnull);
+        shardInfo[rowIndex][2]= DatumGetInt64(nodeId);
 
-        char *maxValueString= PQgetvalue(result, rowIndex, 4);
-		uint64 shardMaxValue = strtou64(maxValueString, NULL, 10);
+        char *shardMinVal = SPI_getvalue(tuple,tupletable->tupdesc,2);
+        shardInfo[rowIndex][3]= strtoi64(shardMinVal,NULL,10);
 
-        ereport(LOG, (errmsg("Shard ID: %d,ShardMinValue: %d, ShardMaxValue: %d , totalSize: %d , nodeId: %d", shardId,shardMinValue,shardMaxValue,totalSize,nodeId)));
+        char *shardMaxVal = SPI_getvalue(tuple,tupletable->tupdesc,3);
+        shardInfo[rowIndex][4]= strtoi64(shardMaxVal,NULL,10);
 
-	}
+        ereport(LOG, (errmsg("Shard ID: %ld,ShardMinValue: %ld, ShardMaxValue: %ld , totalSize: %ld , nodeId: %ld", shardInfo[rowIndex][0],shardInfo[rowIndex][3],shardInfo[rowIndex][4],shardInfo[rowIndex][1],shardInfo[rowIndex][2])));
+           
 
+        
+	}
+    for (int rowIndex = 0; rowIndex < rowCount; rowIndex++)
+    {
+        char *splitquery = GetSplitQuery(shardInfo[rowIndex][3],shardInfo[rowIndex][4],shardInfo[rowIndex][2],shardInfo[rowIndex][0]);
+        ereport(LOG, (errmsg(splitquery))); 
+        StringInfoData buf = { 0 };
+	    initStringInfo(&buf);
+        appendStringInfo(&buf, splitquery);
+        int32 nodesInvolved[] = { 0 };
+		Oid superUserId = CitusExtensionOwner();
+		BackgroundTask *task = ScheduleBackgroundTask(jobId, superUserId, buf.data, 0,
+													  NULL, 0, nodesInvolved);    
+        
+    }
 
+    SPI_freetuptable(tupletable);
+    SPI_finish();
+    
+    
+    
 }
 
 Datum
 citus_auto_shard_split_start(PG_FUNCTION_ARGS){
-   // Citus_Table_List();
-    Citus_get_Data_by_view();
+    Citus_get_shard_data();
```

Comment:
only the UDF name should be in lowercase seperated by underscores.
All the other functions should be in "CamelCase" like this

Reply:
Fixed it